java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -127,7 +128,15 @@ open class KotlinFileExtractor(
                 }
             }
             extractStaticInitializer(file, { extractFileClass(file) })
-            CommentExtractor(this, file, tw.fileId).extract()
+            val psiCommentsExtracted = CommentExtractorPSI(this, file, tw.fileId).extract()
+            val lighterAstCommentsExtracted = CommentExtractorLighterAST(this, file, tw.fileId).extract()
```

Comment:
This feels somewhat odd. Can't we decide ahead of time which variant we would need to try?

Reply:
I can't immediately see an easy way. In any case, I think that just extracting whatever is there, and warning if it's unexpected, is reasonable; especially when other plugins could potentially be doing something unexpected..